### PR TITLE
fix(dynamite): json schema reference resolving

### PR DESCRIPTION
**Pull request by Leptopoda**
I did not find a better solution yet.
In theory this already allows us to fetch schemas fro other sources like local filesystem or remote servers.
I did not observe any performance penalty.

Towards: #933

### CODE REVIEW

File: packages/dynamite/dynamite/lib/src/models/openapi/schema.dart [hunk 1]
```diff
@@ -16,8 +17,37 @@ abstract class Schema implements Built<Schema, SchemaBuilder> {
...
+  @BuiltValueField(wireName: r'$id')
+  Uri? get id;
+
   @BuiltValueField(wireName: r'$ref')
-  String? get ref;
+  Uri? get ref;
+
+  Schema buildRef(Map<String, dynamic> json) {
+    if (ref == null) {
+      throw StateError('Referenced schema can only be built when a ref is present');
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/openapi/schema.dart:28]:
```suggestion
      throw StateError(r'Referenced schema can only be built when a $ref is present');
```

File: packages/dynamite/dynamite/lib/src/models/openapi/schema.dart [hunk 2]
```diff
@@ -16,8 +17,37 @@ abstract class Schema implements Built<Schema, SchemaBuilder> {
...
   static Serializer<Schema> get serializer => _$schemaSerializer;
 
+  @BuiltValueField(wireName: r'$id')
+  Uri? get id;
+
   @BuiltValueField(wireName: r'$ref')
-  String? get ref;
+  Uri? get ref;
+
+  Schema buildRef(Map<String, dynamic> json) {
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/openapi/schema.dart:26]:
```suggestion
  Schema resolveRef(Map<String, dynamic> json) {
```
I think this better reflects what it does. The ref isn't built, the ref is resolved and then the schema for it is built

**Review by provokateurin — CHANGES_REQUESTED** [review]
Could this have some tests? I see that it is not breaking anything because none of the generated code has changed, but that is no guarantee it won't break in the future.

**Comment by Leptopoda** [comment]
We already had the e2e tests for this but I added some directly dynamite nevertheless

**Review by provokateurin — APPROVED** [review]

**Comment by provokateurin** [comment]
Commit message has a typo and should be `fix(dynamite): json schema reference resolving`

**Pull request merged**